pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java

```diff
@@ -162,7 +165,10 @@ public PulsarAdmin(String serviceUrl,
 
         this.client = clientBuilder.build();
 
-        this.serviceUrl = serviceUrl;
+        this.serviceNameResolver = new PulsarServiceNameResolver();
+        this.serviceNameResolver.updateServiceUrl(serviceUrl);
+
+        this.serviceUrl = this.serviceNameResolver.resolveHostUri().toString();
```

Comment:
This will only pick one host (for the `PulsarAdmin` instance) and it will never retry if that host is not reachable

Reply:
Thanks for review. You are right. That's only support choose a random host for service. I try to add a retry that's need wrapper almost all http method in jersey. Then i found all method invoke by `new JerseyInvocation(this)).invoke(responseType)`, but that is private so i can't extend. If you have any suggestion feel free to let me know.

Reply:
Take a look at `AsyncHttpConnector`. In the `apply()` method we should be able to replace the host with one provided by the serviceNameResolver.

Reply:
Thank you! I will try.